script.js

```diff
@@ -1,4 +1,300 @@
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //game mode for end of game
+  if (gameMode === "finished") {
+    return "Game is finished, please refresh the page to play again";
+  }
+  //game mode for user name input
+  if (gameMode === "waiting for name") {
+    userName = input;
+    gameMode = "waiting for normal or reverse game mode";
+    return (
+      "Hello " +
+      userName +
+      "! To start off please decide whether you want to play the normal or reverse mode, by typing in one of these two options. Good luck!"
+    );
+  }
+  //game mode for reverse/normal game
+  if (gameMode === "waiting for normal or reverse game mode") {
+    gameOption = input;
+    console.log("gameoption");
+    console.log(gameOption);
+    if (input === "reverse" || input === "normal") {
+      gameMode = "waiting for korean mode";
+      return (
+        "Hello " +
+        userName +
+        "! You have chosen the " +
+        gameOption +
+        " option of the game. Please now decide whether you want to play the korean version of the game or the normal one!"
+      );
+    } else
+      return "Try again, you need to chose between normal and reverse game play!";
+  }
+  //game mode for korean mode
+  if (gameMode === "waiting for korean mode") {
+    koreanMode = input;
+    console.log("koreanMode");
+    console.log(koreanMode);
+    if (input === "korean" || input === "normal") {
+      gameMode = "ready to go";
+      return (
+        "Hello " +
+        userName +
+        "! You have chosen the " +
+        koreanMode +
+        " option of the game. Good luck!"
+      );
+    } else
+      return "Try again, you need to chose between korean and normal game play!";
+  }
+  //normal mode not korean
+  if (gameMode === "ready to go" && koreanMode === "normal") {
+    console.log("gameoption");
+    console.log(gameOption);
+    countGames = countGames + 1;
+    differenceValueOutput = difference(input);
+    var myOutputValue =
+      "Try again: you need to chose between scissors, paper and rock!";
+    if (gameOption === "reverse") {
+      if (differenceValueOutput == -1) {
+        myOutputValue = "You lose";
+      }
+      if (differenceValueOutput == 1) {
+        myOutputValue = "You win";
+      }
+      if (differenceValueOutput == 2) {
+        myOutputValue = "You lose";
+      }
+      if (differenceValueOutput == -2) {
+        myOutputValue = "You win";
+      }
+      if (differenceValueOutput == 0) {
+        myOutputValue = "Draw...";
+      }
+    } else {
+      if (differenceValueOutput == 1) {
+        myOutputValue = "You lose";
+      }
+      if (differenceValueOutput == -1) {
+        myOutputValue = "You win";
+      }
+      if (differenceValueOutput == -2) {
+        myOutputValue = "You lose";
+      }
+      if (differenceValueOutput == 2) {
+        myOutputValue = "You win";
+      }
+      if (differenceValueOutput == 0) {
+        myOutputValue = "Draw...";
+      }
```

Comment:
Instead of having this many "if" , you should use a Boolean  logical operator "||".